examples/api/lib/widgets/linked_text/linked_text.0.dart

```diff
@@ -0,0 +1,80 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+
+// This example demonstrates using LinkedText to make URLs open on tap.
+
+void main() {
+  runApp(const LinkedTextApp());
+}
+
+class LinkedTextApp extends StatelessWidget {
+  const LinkedTextApp({
+    super.key,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      title: 'Flutter Demo',
+      theme: ThemeData(
+        primarySwatch: Colors.blue,
+      ),
+      home: const MyHomePage(title: 'Flutter Link Demo'),
+    );
+  }
+}
+
+class MyHomePage extends StatelessWidget {
+  const MyHomePage({
+    super.key,
+    required this.title,
+  });
+
+  final String title;
+  static const String _text = 'Check out https://www.flutter.dev, or maybe just flutter.dev or www.flutter.dev.';
+
+  void _handleTapUri(BuildContext context, String urlString) {
+    final Uri? uri = Uri.tryParse(urlString);
+    if (uri == null) {
+      throw Exception('Failed to parse $urlString.');
```

Comment:
API-wise this is a bit unpleasant. Since the `LinkedText` widget is essentially a parser, can we give the API user a `Uri` instead of the raw string?

Reply:
Sometimes the matched string might not be a Uri, for example the Twitter handle example in linked_text.1.dart.  I think I could do it by adding a type parameter to LinkedText and LinkTapCallback if you think it's worth it?

Reply:
each constructor can take different onTapCallback, where underneath wire to a generic `ValueChanged<String>`, for the one that automatically parse url you can have a `ValueChanged<Uri>`.

Reply:
I couldn't figure out a good way to approach this idea.  Do you mean LinkedText should have a type parameter, or are you thinking the default constructor should handle Uris by default and another constructor should handle strings?

Also, is the ValueNotifier a replacement for the onTap callback?

Reply:
you can have something like this

```dart
LinkedText._raw(ValueChanged<String> onTap);

LinkText(ValueChanged<Uri> onLinkTap) : super._raw(onTap: _getOnTap(onLinkTap));

static ValueChanged<String> _getOnTap(ValueChanged<Uri> onLinkTap) {
  return (String text) {
    onLinkTap(Uri.parse(text));
  };
}
```

> Also, is the ValueNotifier a replacement for the onTap callback?

I don't know the answer, but I do know the draw back for using ValueNotifier is that its life cycle also need to be taking care of. I think if this is a public api, using `onTap` will be easier for the developer and less error prompt.

Reply:
on I think I know where the confusing is coming from. I meant ValueChanged not ValueNotifier
```dart
typedef ValueChanged<T> = void Function(T value);
```

Reply:
Ah you're right, I totally read `ValueChanged` as `ValueNotifier`.  Thanks for clarifying.

I've made the default `LinkedText` constructor work with Uris only, and I've brought back the `.regExp` constructor for non-Uri use cases.  Thanks for the suggestion.